### PR TITLE
Support TPU Topology field for 1VM TPU

### PR DESCRIPTION
**Pull request by vivianrwu**
Add support for GKE 1VM TPU. 
- Add `tpu_topology` field in NodePool schema to support TPU machines.

Fixes hashicorp/terraform-provider-google/issues/15050

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added field `tpu_topology` under `placement_policy` in resource `google_container_node_pool`
```

### CODE REVIEW

**Comment by vivianrwu** [comment]
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8271-old..auto-pr-8271) ( 3 files changed, 47 insertions(+), 49 deletions(-)) Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8271-old..auto-pr-8271) ( 5 files changed, 124 insertions(+), 51 deletions(-)) TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8271-old..auto-pr-8271) ( 2 files changed, 50 insertions(+), 49 deletions(-))
> 
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_container_cluster` (1 total tests) Untested fields: `node_pool.placement_policy.tpu_topology`
> 
> Please add acceptance tests which include these fields.

@hao-nan-li Could you clarify what this means? I am not sure why the `google_container_cluster` resource would need to be tested with the `tpu_topology` field since the field is at the node pool level.

**Comment by hao-nan-li** [comment]
> > Hi there, I'm the Modular magician. I've detected the following information about your changes:
> > ## Diff report
> > Your PR generated some diffs in downstreams - here they are.
> > Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-8271-old..auto-pr-8271) ( 3 files changed, 47 insertions(+), 49 deletions(-)) Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-8271-old..auto-pr-8271) ( 5 files changed, 124 insertions(+), 51 deletions(-)) TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-8271-old..auto-pr-8271) ( 2 files changed, 50 insertions(+), 49 deletions(-))
> > ## Missing test report
> > Your PR includes resource fields which are not covered by any test.
> > Resource: `google_container_cluster` (1 total tests) Untested fields: `node_pool.placement_policy.tpu_topology`
> > Please add acceptance tests which include these fields.
> 
> @hao-nan-li Could you clarify what this means? I am not sure why the `google_container_cluster` resource would need to be tested with the `tpu_topology` field since the field is at the node pool level.

I think you can ignore this for now.

File: mmv1/third_party/terraform/go.mod.erb [hunk 1]
```diff
@@ -25,12 +25,12 @@ require (
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/mitchellh/hashstructure v1.1.0
 	github.com/sirupsen/logrus v1.8.1
-	golang.org/x/net v0.10.0
-	golang.org/x/oauth2 v0.8.0
-	google.golang.org/api v0.128.0
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20230530153820-e85fd2cbaebc
-	google.golang.org/grpc v1.55.0
-	google.golang.org/protobuf v1.30.0
+	golang.org/x/net v0.12.0
```
[the remark this answers is not in the document]
Comment (vivianrwu) [reply mmv1/third_party/terraform/go.mod.erb:28]: They are not required for adding the TPU Topology field. They updated when I ran `go mod tidy`. I can revert this to only change the `golang.org/api` version if you would like.  Let me know.

[the remark this answers is not in the document]
Comment (hao-nan-li) [reply mmv1/third_party/terraform/go.mod.erb:28]: Thanks for the confirmation, I'd prefer to have them in 2 different PRs.

[the remark this answers is not in the document]
Comment (vivianrwu) [reply mmv1/third_party/terraform/go.mod.erb:28]: For clarification, do you mean that you would want the updated dependencies in a different PR and I should just update `golang.org/api` in this one?

[the remark this answers is not in the document]
Comment (vivianrwu) [reply mmv1/third_party/terraform/go.mod.erb:28]: Sounds good. I'll work on them

[the remark this answers is not in the document]
Comment (vivianrwu) [reply mmv1/third_party/terraform/go.mod.erb:28]: @hao-nan-li I took a look at the dependencies and all of these updates are required to add the `tpu_topology` field. 
`tpu_topology` is added here in google api version v0.130 - https://github.com/googleapis/google-api-go-client/pull/2051/files#diff-d9ab87713cc1f8facb5476cea4b9afa869a352cd21b6b894fa3eba305cd47dd1R5947 

See https://github.com/googleapis/google-api-go-client/blob/v0.130.0/go.mod for its dependencies and their required versions. Let me know if you have any further questions.

[the remark this answers is not in the document]
Comment (hao-nan-li) [reply mmv1/third_party/terraform/go.mod.erb:28]: How about the other dependencies below?

[the remark this answers is not in the document]
Comment (vivianrwu) [reply mmv1/third_party/terraform/go.mod.erb:28]: Yes, I can outline them: 
Dependencies of google.golang.org/api (ref https://github.com/googleapis/google-api-go-client/blob/v0.130.0/go.mod): 
- golang.org/x/net 
- golang.org/x/oauth2
- google.golang.org/genproto/googleapis/rpc
- google.golang.org/grpc
- google.golang.org/protobuf
- github.com/googleapis/enterprise-certificate-proxy
- github.com/googleapis/gax-go/v2
- golang.org/x/crypto
- golang.org/x/sys
- golang.org/x/text 

Dependencies of google.golang.org/grpc (dependency of google.golang.org/api) (ref https://github.com/grpc/grpc-go/blob/v1.56.2/go.mod): 
- github.com/cncf/xds/go
- github.com/envoyproxy/go-control-plane
- github.com/envoyproxy/protoc-gen-validate

**Comment by hao-nan-li** [comment]
Looks like the added test failed due to permission issues, could you confirm this test passed locally?

**Comment by vivianrwu** [comment]
> Looks like the added test failed due to permission issues, could you confirm this test passed locally?

Yes I can confirm that the test passes locally.

**Comment by vivianrwu** [comment]
Added a skip_vcr on TestAccContainerNodePool_tpuTopology due to the potential flakiness of testing that will come with TPU resources. Please reference internal bug 285016550 comment 15 for further reasoning and proof of acceptance test working and passing local testing.

**Comment by hao-nan-li** [comment]
b/285016550 for reference.

**Review by hao-nan-li — APPROVED** [review]

**Pull request merged**